docs/book/provider_implementations/overview.md

```diff
@@ -0,0 +1,43 @@
+# Overview
+
+In order to demonstrate how to develop a new Cluster API provider we will use 
+`kubebuilder` to create an example provider. For more information on `kubebuilder`
```

Comment:
This should likely call out that this is an example *infrastructure* provider. Implementing a bootstrapping provider would potentially be different.